hpy/tools/autogen/conf.py

```diff
@@ -0,0 +1,111 @@
+# Defines parameters for the code generation
```

Comment:
These constants were pulled from other files into this central location.

Reply:
This might not be necessary right now but maybe the configuration could use a schema. This will make things easier as this configuration file will grow in size and options.

In the meantime, please document how to use this configuration file in a docstring.

Reply:
> In the meantime, please document how to use this configuration file in a docstring.

Do you have suggestion what else to comment on? I've added a comment to each constant in this file.

Reply:
Since these are dictionaries which are untyped we need to document the schema.
We could use https://docs.python.org/3/library/typing.html#typing.TypedDict or dataclasses instead of dictionaries.

Reply:
We should also include documentation regarding this in the contribution guide.

Reply:
It seems that we do not have any extensive contribution guide or documentation? No documentation file in the repo mentions `make autogen`, the closest is https://github.com/hpyproject/hpy/blob/master/CONTRIBUTING.md but it is very basic. 

I agree that adding more extensive developer oriented documentation would be good, but I think it is out of the scope of this PR. The same applies to the suggestion to use schema. Currently in the master the "configuration" is scattered over different files, so putting it in one file is IMO improvement already. We can create a followup issue for improving this even further.